apps/src/templates/studioHomepages/CourseBlocks.jsx

```diff
@@ -0,0 +1,203 @@
+    $('#widgets').appendTo(ReactDOM.findDOMNode(this.refs.widgets)).show();
+    $('#gamelab').appendTo(ReactDOM.findDOMNode(this.refs.gamelab)).show();
+    $('#weblab').appendTo(ReactDOM.findDOMNode(this.refs.weblab)).show();
+  },
+
+  render() {
+    return (
+      <div>
+        <div className="row">
+          <ProtectedStatefulDiv ref="applab"/>
+          <ProtectedStatefulDiv ref="widgets"/>
+        </div>
+        <br/>
+        <br/>
+        <div className="row">
+          <ProtectedStatefulDiv ref="gamelab"/>
+          <ProtectedStatefulDiv ref="weblab"/>
+        </div>
+      </div>
+    );
+  }
+});
+
+export const CourseBlocksHoc = React.createClass({
+  propTypes: {
+    rowCount: React.PropTypes.number.isRequired
+  },
+
+  componentDidMount() {
+    $('#minecraft').appendTo(ReactDOM.findDOMNode(this.refs.minecraft)).show();
+    $('#starwars').appendTo(ReactDOM.findDOMNode(this.refs.starwars)).show();
+    $('#frozen').appendTo(ReactDOM.findDOMNode(this.refs.frozen)).show();
+    $('#hourofcode').appendTo(ReactDOM.findDOMNode(this.refs.hourofcode)).show();
+    $('#flappy').appendTo(ReactDOM.findDOMNode(this.refs.flappy)).show();
+    $('#infinity').appendTo(ReactDOM.findDOMNode(this.refs.infinity)).show();
+    $('#playlab').appendTo(ReactDOM.findDOMNode(this.refs.playlab)).show();
+    $('#artist').appendTo(ReactDOM.findDOMNode(this.refs.artist)).show();
+  },
+
+  render() {
+    return (
+      <div>
+        <div className="row">
+          <ProtectedStatefulDiv ref="minecraft"/>
+          <ProtectedStatefulDiv ref="starwars"/>
+          <ProtectedStatefulDiv ref="frozen"/>
+          <ProtectedStatefulDiv ref="hourofcode"/>
+        </div>
+        <br/>
+        <br/>
+        {this.props.rowCount > 1 && (
+          <div className="row">
+            <ProtectedStatefulDiv ref="flappy"/>
+            <ProtectedStatefulDiv ref="infinity"/>
+            <ProtectedStatefulDiv ref="playlab"/>
+            <ProtectedStatefulDiv ref="artist"/>
+          </div>
+        )}
+      </div>
+    );
+  }
+});
+
+export const CourseBlocksAll = React.createClass({
+  propTypes: {
+    isEnglish: React.PropTypes.bool.isRequired
+  },
+
+  componentDidMount() {
+    $('.csf-courses-header').appendTo(ReactDOM.findDOMNode(this.refs.csfCoursesHeader)).show();
+    $('.tools-header').appendTo(ReactDOM.findDOMNode(this.refs.toolsHeader)).show();
+    $('.hoc-courses-header').appendTo(ReactDOM.findDOMNode(this.refs.hocCoursesHeader)).show();
+  },
+
+  render() {
+    return (
+      <div>
+        {!this.props.isEnglish && (
+          <ProtectedStatefulDiv ref="csfCoursesHeader"/>
+        )}
+        <CourseBlocksCsf isEnglish={this.props.isEnglish}/>
+
+        <ProtectedStatefulDiv ref="toolsHeader"/>
+        <CourseBlocksTools/>
+
+        <br/>
+        <br/>
+
+        <ProtectedStatefulDiv ref="hocCoursesHeader"/>
+        <CourseBlocksHoc rowCount={2}/>
+      </div>
+    );
+  }
+});
```

Comment:
I personally don't love having lots of different components in the same file. It might be okay here, but I'd like to at least see comments describing what the different components are.

Should we also have some basic unit test coverage of these components?